students/kristianjf/lesson09/mailroom_5.py

```diff
@@ -201,21 +162,75 @@ def name_response_valid(self, full_name):
         else:
             return full_name
 
-def main():
+#Function to Process Menu Options
+def menu(options_dict, **kwargs):
+    '''
+    The script should prompt the user (you) to choose from a menu of 3 actions: \
+    “Send a Thank You”, “Create a Report” or “quit”)
+    '''
+    options = [option[0:2] for option in enumerate(options_dict.keys(), 1)]
+    while True:
+        print('Please select a number from the list of the following options: \n')
+        for option in options:
+            print(option)
+        response = menu_response_valid(options)
+        response_selection = options[response-1][1]
+        options_dict[response_selection](**kwargs)
+
+#Main Menu
+def main_menu(**kwargs):
+    'Create Main Menu'
+    program_options_dict = {'Send a Thank You': thank_you_menu, \
+                            'Create a Report': kwargs['donor_handler'].create_a_report, \
+                            'Send Letters to Everyone': kwargs['donor_handler'].send_letters, \
+                            'quit': quit_menu}
+    menu(program_options_dict, **kwargs)
+
+#Thank You Menu
+def thank_you_menu(**kwargs):
+    '''
+    Sending a Thank You
+    If the user (you) selects ‘Send a Thank You’, prompt for a Full Name.
+    If the user types ‘list’, show them a list of the donor names and re-prompt
+    '''
+    program_options_ty = {'Send a Thank You': kwargs['donor_handler'].thank_you, \
+                        'Action: List': kwargs['donor_handler'].list_donors, \
```

Comment:
I suppose you could have just passed the donor_handler object as a usual argument, not as kwargs, but I may be nitpicking anyway... :) Just wondering if there's any special reason for you to choose kwargs rather than a normal argument?

Reply:
agreed -- better to pass the dict in as a regular parameter.

Reply:
Thanks for the feedback. In retrospect, passing kwargs was not necessary. I believe I previously attempted to pass kwargs outside the class. Now looking at the code, kwargs for the menu may be an unnecessary complication.

Thanks!